pkg/routeagent_driver/handlers/ovn/gateway_route_controller.go

```diff
@@ -0,0 +1,150 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ovn
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/watcher"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type GatewayRouteController struct {
+	gatewayRouteWatcher watcher.Interface
+	connectionHandler   *ConnectionHandler
+	mutex               sync.Mutex
+	remoteSubnets       sets.Set[string]
+	stopCh              chan struct{}
+	mgmtIP              string
+}
+
+func NewGatewayRouteController(config *watcher.Config, connectionHandler *ConnectionHandler,
```

Comment:
It's safer to pass `config` by value since it's mutated.

```suggestion
func NewGatewayRouteController(config watcher.Config, connectionHandler *ConnectionHandler,
```

Reply:
Done.

Reply:
Reverted back since linter is throwing an error like object is big and consider using pointer.

Reply:
You can either disable the linter via `//nolint:gocritic // Ignore hugeParam` (which we've done elsewhere) or make an internal copy of it and mutate the copy. The bottom line is not to modify the formal param.

Reply:
Done.